tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -455,6 +439,19 @@ def test_dataclass_transformer():
     assert t.metadata is None
 
 
+def test_dataclass_to_python_value():
+    ctx = FlyteContext.current_context()
+    dataclass_tf = DataclassTransformer()
+    dict_tf = DictTransformer()
+    o = TestStructC(s=InnerStruct(a=5, b=None, c=[1, 2, 3]), m={"a": 5})
+    literal_type = dataclass_tf.get_literal_type(TestStructC)
+    lv = dataclass_tf.to_literal(ctx, o, TestStructC, literal_type)
+    expected_python_type = dict_tf.guess_python_type(literal_type)
+    ot = dataclass_tf.to_python_value(ctx=ctx, lv=lv, expected_python_type=expected_python_type)
+    final_object = dataclass_from_dict(TestStructC, asdict(ot))  # type: ignore
+    assert final_object == o
```

Comment:
cc @kumare3 It could fix the error in https://github.com/flyteorg/flytesnacks/pull/424.
`ot` is like flyte remote execution output, and we could convert to user-defined dataclass by using `dataclass_from_dict`